app/src/lib/stores/app-store.ts

```diff
@@ -1424,13 +1423,22 @@ export class AppStore {
     const state = this.getRepositoryState(repository)
     const gitStore = this.getGitStore(repository)
 
+    this.updateBranchesState(repository, state => {
+      return {
+        currentPullRequest: null,
```

Comment:
It seems like this would cause a bug like https://github.com/desktop/desktop/issues/3536, where we clear it, render, find the associate PR again, render it again.

Reply:
So this is being cleared, to make sure `currentPullRequest` doesn't hold values from the last branch that had a PR. This ensures that branches with no PRs don't retain the PR badge.

Reply:
Yeah, this came up when I was pairing with Will on the issue yesterday. When `git checkout` to switch branches takes a while, leaving the old PR badge around felt like a smell, so unsetting it felt like a safe bet while this whole process was happening.

I'll see how the current branch feels around this - it was noticeable with `microsoft/vscode`...

Reply:
Yeah, I can see how that'd be less than great. If this is a problem, can we clear it only when the branch changes, instead of every time we refresh?